src/urllib3/response.py

```diff
@@ -1073,6 +1076,10 @@ def readable(self) -> bool:
         return True
 
     def shutdown(self) -> None:
+        if self._released_conn:
```

Comment:
```suggestion
        if not self._connection:
```
There's no need for an extra attribute here.

Reply:
@D-stefaang Thank you for the review. I have removed the extra attribute now.